cpp/ql/test/query-tests/Critical/MissingCheckScanf/test.cpp

```diff
@@ -94,22 +116,43 @@ int main()
 		}
 	}
 
+	{
+		int r;
+		int i;
+
+		r = scanf("%d", &i); // GOOD
+
+		if (r >= 1)
+		{
+			use(i);
+		}
+	}
+
 	{
 		bool b;
 		int i;
 
-		b = scanf("%d", &i); // GOOD
+		b = scanf("%d", &i); // BAD [NOT DETECTED]: scanf can return EOF (boolifies true)
 
 		if (b >= 1)
 		{
 			use(i);
 		}
 	}
 
+	{
+		bool b;
+		int i;
+
+		b = scanf("%d", &i); // BAD [NOT DETECTED]
+
+		use(i);
+	}
+
 	{
 		int i, j;
 
-		if (scanf("%d %d", &i) >= 2) // GOOD
+		if (scanf("%d %d", &i) >= 2) // GOOD: `j` is not a scanf arg, so out of scope of MissingCheckScanf
```

Comment:
I think I typo'd this, the intended test case was:
```
if (scanf("%d %d", &i, &j) >= 2) // GOOD
```

Reply:
I added a separate test case for this :)